File: mealie/schema/events.py
from datetime import datetime
from enum import Enum
from typing import Optional

from fastapi_camelcase import CamelModel
from pydantic import Field


class EventCategory(str, Enum):
    general = "general"
    recipe = "recipe"
    backup = "backup"
    scheduled = "scheduled"
    migration = "migration"
    group = "group"
    user = "user"


class Event(CamelModel):
    id: Optional[int]
    title: str
    text: str
    time_stamp: datetime = Field(default_factory=datetime.now)
    category: EventCategory = EventCategory.general

    class Config:
        orm_mode = True


class EventsOut(CamelModel):
    total: int
    events: list[Event]


class TestEvent(CamelModel):
    id: Optional[int]
    test_url: Optional[str]

File: frontend/api/class-interfaces/events.ts
import { BaseAPI } from "./_base";

export type EventCategory = "general" | "recipe" | "backup" | "scheduled" | "migration" | "group" | "user";

export interface Event {
  id?: number;
  title: string;
  text: string;
  timeStamp?: string;
  category?: EventCategory & string;
}

export interface EventsOut {
  total: number;
  events: Event[];
}

const prefix = "/api";

const routes = {
  aboutEvents: `${prefix}/about/events`,
  aboutEventsNotifications: `${prefix}/about/events/notifications`,
  aboutEventsNotificationsTest: `${prefix}/about/events/notifications/test`,

  aboutEventsId: (id: number) => `${prefix}/about/events/${id}`,
  aboutEventsNotificationsId: (id: number) => `${prefix}/about/events/notifications/${id}`,
};

export class EventsAPI extends BaseAPI {
  /** Get event from the Database
   */
  async getEvents() {
    return await this.requests.get<EventsOut>(routes.aboutEvents);
  }

  /** Get event from the Database
   */
  async deleteEvents() {
    return await this.requests.delete(routes.aboutEvents);
  }

  /** Delete event from the Database
   */
  async deleteEvent(id: number) {
    return await this.requests.delete(routes.aboutEventsId(id));
  }
  /** Get all event_notification from the Database
   */
}
